distributed/diagnostics/plugin.py

```diff
@@ -164,7 +167,7 @@ def add_worker(self, scheduler: Scheduler, worker: str) -> None | Awaitable[None
         """
 
     def remove_worker(
-        self, scheduler: Scheduler, worker: str
+        self, scheduler: Scheduler, worker: str, *, stimulus_id: str
     ) -> None | Awaitable[None]:
```

Comment:
```suggestion
    def remove_worker(
        self, scheduler: Scheduler, worker: str, stimulus_id: str, **kwargs
    ) -> None | Awaitable[None]:
```

I suggest to not force the keyword and allow kwargs for future compat reasons, see https://github.com/dask/distributed/pull/7974#issuecomment-1632295809

Reply:
Forcing the keyword while adding `**kwargs` would make a method with the signature `MySchedulerPlugin.remove_worker(self, scheduler, worker, **kwargs)` fully forward-compatible, so I'd suggest sticking with it unless we also want to require `*args` in the signature.